src/Main.elm

```diff
@@ -2,21 +2,64 @@ module Main exposing (main)
 
 import Browser
 import Data
+import Either exposing (Either(..))
 import Html exposing (Html)
 import Html.Attributes as Attrs
+import Html.Events as Evts
+import Json.Decode as Json
+import Json.Encode as Json
+import Scrive.Data.Address as CD
+import Scrive.Data.ContactDetails as CD
+import Scrive.Data.RetentionPolicy as RP
+import Scrive.Data.Tag as Tag exposing (ArchivedTag, SomeTag, Tag)
+import Scrive.Data.UserGroup as UG exposing (UserGroup)
+import Scrive.Form.Error as FE
+import Scrive.Form.Field as BelongsTo exposing (BelongsTo(..))
+import Scrive.Form.Impl.ContactDetails as ContactForm
+import Scrive.Form.Impl.RetentionPolicy as RetentionPolicy
+import Scrive.Form.Impl.Tags as TagsForm
+import Scrive.Form.Validator exposing (addressValidator, tagValidator)
+import Set
+import Style as Style
+import Validate as V exposing (Valid, Validator)
+import Validate.Extra as V
 
 
 
 ---- MODEL ----
 
 
+type EditFocus
+    = FocusTag TagsForm.TagInProgress
+    | FocusPolicyAdd RP.DataRetentionPolicy
+    | FocusPolicyEdit ( RP.DataRetentionPolicy, Int )
+    | FocusContactsEdit ( CD.Field, String )
+    | NotEditing
+
+
 type alias Model =
-    {}
+    { userGroup : Result Json.Error UserGroup
+    , editFocus : EditFocus
+    , validationErrors : List FE.Error
+    , viewMode : ViewJson -- TODO: remove, only for debugging
```

Comment:
Does it make sense to have `editFocus`, `validationErrors` and `viewMode` when `userGroup == Err ...`?

Reply:
Yes, having to juggle that `Result ... UserGroup` all the time was a bit of a code smell.